scripts/release.sh

```diff
@@ -0,0 +1,568 @@
+        fi
+
+        # Check for duplicates
+        local found_duplicate=false
+        for seen in "${seen_modules[@]}"; do
+            if [[ "$seen" == "$dir" ]]; then
+                found_duplicate=true
+                break
+            fi
+        done
+
+        if [[ "$found_duplicate" == false ]]; then
+            modules+=("$dir")
+            seen_modules+=("$dir")
+        fi
+    done < <(find "$REPO_ROOT" -name "go.mod" -print0)
+
+    printf '%s\n' "${modules[@]}"
+}
+
+# Get the current global version from all modules
+get_current_global_version() {
+    local all_tags=()
+
+    # Collect all version tags from all modules (ignore prerelease format warnings)
+    while IFS= read -r tag; do
+        if [[ "$tag" =~ v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+            all_tags+=("$tag")
+        fi
+    done < <(git tag -l)
+
+    if [[ ${#all_tags[@]} -eq 0 ]]; then
+        echo "v0.0.0"
+        return
+    fi
+
+    # Sort all tags and get the highest version
+    local highest_tag=$(printf '%s\n' "${all_tags[@]}" | sort -V | tail -n1)
+
+    # Extract just the version part (remove module prefix if present)
+    local version=$(echo "$highest_tag" | sed 's/.*\(v[0-9][^-]*\)/\1/')
+
+    echo "$version"
+}
+
+# Increment version based on type
+increment_version() {
+    local current_version="$1"
+    local bump_type="$2"
+
+    # Remove 'v' prefix for processing
+    local version=$(echo "$current_version" | sed 's/^v//')
+
+    # Handle prerelease versions (both 1-digit and 2-digit for parsing existing versions)
+    if [[ "$version" =~ -prerelease[0-9]+$ ]]; then
+        version=$(echo "$version" | sed 's/-prerelease[0-9]*$//')
+    fi
+
+    # Split version into components
+    IFS='.' read -ra VERSION_PARTS <<< "$version"
+    local major=${VERSION_PARTS[0]:-0}
+    local minor=${VERSION_PARTS[1]:-0}
+    local patch=${VERSION_PARTS[2]:-0}
+
+    case "$bump_type" in
+        major)
+            major=$((major + 1))
+            minor=0
+            patch=0
+            ;;
+        minor)
+            minor=$((minor + 1))
+            patch=0
+            ;;
+        patch)
+            patch=$((patch + 1))
+            ;;
+    esac
+
+    echo "${major}.${minor}.${patch}"
+}
+
+# Get next prerelease version globally (strict 2-digit format)
+get_next_prerelease() {
+    local base_version="$1"
+
+    # Find existing prerelease tags for this specific version across all modules
+    local all_prerelease_tags=$(git tag -l "*v${base_version}-prerelease*" | sort -V)
+
+    if [[ -z "$all_prerelease_tags" ]]; then
+        echo "${base_version}-prerelease01"
+        return
+    fi
+
+    # Get the latest prerelease tag for this version
+    local latest_prerelease=$(echo "$all_prerelease_tags" | tail -n1)
+
+    # Check if the LATEST prerelease is 1-digit (this is the only one we care about)
+    if [[ "$latest_prerelease" =~ -prerelease[0-9]$ ]] && [[ ! "$latest_prerelease" =~ -prerelease[0-9][0-9]$ ]]; then
```

Comment:
`-prerelease` looks a little off, it threw me a little, but I guess it's a regex without quotes. I am wondering if `shellcheck complains on this bit? if you've tested it and it's fine then np